barretenberg/cpp/src/barretenberg/vm/tests/AvmMini_execution.test.cpp

```diff
@@ -105,30 +110,37 @@ TEST_F(AvmMiniExecutionTests, setAndSubOpcodes)
                                "00000000"; // ret size 0
 
     auto bytecode = hex_to_bytes(bytecode_hex);
-    auto instructions = Execution::parse(bytecode);
+    auto instructions = Deserialization::parse(bytecode);
 
     EXPECT_EQ(instructions.size(), 4);
 
     // SET
     EXPECT_EQ(instructions.at(0).op_code, OpCode::SET);
-    EXPECT_EQ(instructions.at(0).operands.size(), 2);
-    EXPECT_EQ(instructions.at(0).operands.at(0), 47123);
-    EXPECT_EQ(instructions.at(0).operands.at(1), 170);
-    EXPECT_EQ(instructions.at(0).in_tag, AvmMemoryTag::U16);
+
+    auto operands = instructions.at(0).operands;
+    EXPECT_EQ(operands.size(), 3);
+    EXPECT_EQ(std::get<AvmMemoryTag>(operands.at(0)), AvmMemoryTag::U16);
+    EXPECT_EQ(std::get<uint16_t>(operands.at(1)), 47123);
+    EXPECT_EQ(std::get<uint32_t>(operands.at(2)), 170);
 
     // SET
     EXPECT_EQ(instructions.at(1).op_code, OpCode::SET);
-    EXPECT_EQ(instructions.at(1).operands.size(), 2);
-    EXPECT_EQ(instructions.at(1).operands.at(0), 37123);
-    EXPECT_EQ(instructions.at(1).operands.at(1), 51);
-    EXPECT_EQ(instructions.at(1).in_tag, AvmMemoryTag::U16);
+
+    operands = instructions.at(1).operands;
+    EXPECT_EQ(operands.size(), 3);
+    EXPECT_EQ(std::get<AvmMemoryTag>(operands.at(0)), AvmMemoryTag::U16);
+    EXPECT_EQ(std::get<uint16_t>(operands.at(1)), 37123);
+    EXPECT_EQ(std::get<uint32_t>(operands.at(2)), 51);
 
     // SUB
     EXPECT_EQ(instructions.at(2).op_code, OpCode::SUB);
-    EXPECT_EQ(instructions.at(2).operands.size(), 3);
-    EXPECT_EQ(instructions.at(2).operands.at(0), 170);
-    EXPECT_EQ(instructions.at(2).operands.at(1), 51);
-    EXPECT_EQ(instructions.at(2).in_tag, AvmMemoryTag::U16);
+
+    operands = instructions.at(2).operands;
+    EXPECT_EQ(operands.size(), 4);
+    EXPECT_EQ(std::get<AvmMemoryTag>(operands.at(0)), AvmMemoryTag::U16);
+    EXPECT_EQ(std::get<uint32_t>(operands.at(1)), 170);
+    EXPECT_EQ(std::get<uint32_t>(operands.at(2)), 51);
+    EXPECT_EQ(std::get<uint32_t>(operands.at(3)), 1);
 
     auto trace = Execution::gen_trace(instructions, std::vector<FF>{});
```

Comment:
FYI: the std::vector<> type here shouldn't be needed. `{}` should be enough.

(please add `/*parameterName=*/{}` as well if parameter name not clear from variable name or context)

Reply:
I will address this by having a default argument in gen_trace.